homeassistant/components/media_player/hdmi_cec.py

```diff
@@ -26,9 +26,13 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     """Find and return HDMI devices as +switches."""
     if ATTR_NEW in discovery_info:
         _LOGGER.info("Setting up HDMI devices %s", discovery_info[ATTR_NEW])
-        add_entities(CecPlayerDevice(hass, hass.data.get(device),
-                                     hass.data.get(device).logical_address) for
-                     device in discovery_info[ATTR_NEW])
+        entities = []
+        for device in discovery_info[ATTR_NEW]:
+            hdmi_device = hass.data.get(device)
+            entities.append(CecPlayerDevice(
+                hass, hdmi_device, hdmi_device.logical_address,
```

Comment:
We don't need to pass in `hass`. Please remove `hass` from the signature of the entities. `hass` will be set on the entity when it has been added to home assistant.

Reply:
@MartinHjelmare done!